routes/proxy.js

```diff
@@ -258,6 +258,9 @@ module.exports = function (jwtMiddleware) {
      */
     router.all('/:prefix*', function (req, res, next) {
         var prefix = req.params.prefix;
+        var hostHeader = req.protocol + '://' + req.get('host') + req.originalUrl;
```

Comment:
I think it's better to check if this header is already set (A2 is behind a proxy like Nginx as well) before overriding it. Only setting it when this header has not been configured previously...

```JavaScript

if (! HEADER_x-forwarded-host) {
    var hostHeader = req.protocol + '://' + req.get('host') + req.originalUrl;
    hostHeader = hostHeader.slice(-1) === '/' ? hostHeader : hostHeader + '/';
    req.headers['x-forwarded-host'] = hostHeader;
}
```